specification.md

```diff
@@ -1075,9 +1077,167 @@ Note that the referenced function definition type in this case must be `rpc`.
+
+And the outer `data` container object must be omitted from the response of successful queries and mutations, leaving only the selection set's specified fields. For exceptions the response must be returned in its entirety.
+
+We'll use the following [schema definition](https://graphql.org/learn/schema/) to show how that would work with both a query and a mutation:
+
+```graphql
+type Query {
+    pets: [Pet]
+    pet(id: Int!): Pet
+}
+
+type Mutation {
+    createPet(pet: PetInput!): Pet
+}
+
+type Treat {
+    id: Int!
+}
+
+type Pet {
+    id: Int!
+    name: String!
+    favoriteTreat: Treat
+}
+
+input PetInput {
+    id: Int!
+    name: String!
+    favoriteTreatId: Int
+}
+```
+
+#### Executing a `Query`
+
+In our workflow definition, we can then use a function definition for the `pet` query field as such:
+
+```json
+{
+  "functions": [
+    {
+      "name": "getOnePet",
+      "operation": "https://example.com/pets/graphql#query#pet",
+      "type": "graphql"
+    }
+  ]
+}
+```
+
+Note that the `operation` property has the following format for the `graphql` type:
+
+```text
+<url_to_graphql_endpoint>#<literal "mutation" or "query">#<mutation_or_query_field>
+```
+
+In order to invoke this query, we would use the following `functionRef` parameters:
+
+```json
+{
+  "refName": "getOnePet",
+  "arguments": {
+    "inputs": {
+      "id": 42
+    },
+    "selectionSet": "{ id, name, favoriteTreat { id } }"
+  }
+}
+```
+
+Which would return the following action state:
+
+```json
+{
+  "pet": {
+    "id": 42,
+    "name": "Snuffles",
+    "favoriteTreat": {
+      "id": 9001
+    }
+  }
+}
+```
+
+#### Executing a `Mutation`
+
+Likewise, we would use the following function definition:
+
+```json
+{
+  "functions": [
+    {
+      "name": "createPet",
+      "operation": "https://example.com/pets/graphql#mutation#createPet",
+      "type": "graphql"
+    }
+  ]
+}
+```
+
+With the parameters for the `functionRef`:
+
+```json
+{
+  "refName": "createPet",
+  "arguments": {
+    "inputs": {
+      "pet": {
+        "id": 43,
+        "name":"Sadaharu",
+        "favoriteTreatId": 9001
+      }
+    },
+    "selectionSet": "{ id, name, favoriteTreat { id } }"
+  }
+}
+```
+
+Which would execute the mutation, creating the object and returning it the following data:
+
+```json
+{
+  "pet": {
+    "id": 43,
+    "name": "Sadaharu",
+    "favoriteTreat": {
+      "id": 9001
+    }
+  }
+}
+```
+
+Note you can include [expressions](#Workflow-Expressions) in both both `inputs` and `selectionSet`:
+
+```json
+{
+  "refName": "getOnePet",
+  "arguments": {
+    "inputs": {
+      "id": "${ .petId }"
+    },
+    "selectionSet": "{ id, name, age(useDogYears: ${ .isPetADog }) { dateOfBirth, years } }"
+  }
+}
+```
+
+Expressions must be evaluated before executing the operation.
+
+Also note that at this time subscriptions are not supported due to the workflow model.
```

Comment:
Maybe use something like

Note that GraphQL Subscriptions are not supported at this time.